src/platforms/windows/daemon/dnsutilswindows.cpp

```diff
@@ -0,0 +1,179 @@
+  settings.RegistrationEnabled = false;
+  settings.RegisterAdapterName = false;
+  settings.EnableLLMNR = false;
+  settings.QueryAdapterName = false;
+  settings.ProfileNameServer = nullptr;
+
+  // Configure nameservers for IPv4
+  QString v4resolverstring = v4resolvers.join(",");
+  settings.NameServer = (wchar_t*)v4resolverstring.utf16();
+  DWORD v4result = m_setInterfaceDnsSettingsProcAddr(guid, &settings);
+  if (v4result != NO_ERROR) {
+    logger.error() << "Failed to configure IPv4 resolvers:" << v4result;
+  }
+
+  // Configure nameservers for IPv6
+  QString v6resolverstring = v6resolvers.join(",");
+  settings.Flags |= DNS_SETTING_IPV6;
+  settings.NameServer = (wchar_t*)v6resolverstring.utf16();
+  DWORD v6result = m_setInterfaceDnsSettingsProcAddr(guid, &settings);
+  if (v6result != NO_ERROR) {
+    logger.error() << "Failed to configure IPv6 resolvers" << v6result;
+  }
+
+  return ((v4result == NO_ERROR) && (v6result == NO_ERROR));
+}
+
+constexpr const char* netshFlushTemplate =
+    "interface %1 set dnsservers name=%2 address=none valdiate=no "
+    "register=both\r\n";
+constexpr const char* netshAddTemplate =
+    "interface %1 add dnsservers name=%2 address=%3 validate=no\r\n";
+
+bool DnsUtilsWindows::updateResolversNetsh(
+    const QList<QHostAddress>& resolvers) {
+  QProcess netsh;
+  NET_LUID luid;
+  NET_IFINDEX ifindex;
+  luid.Value = m_luid;
+  if (ConvertInterfaceLuidToIndex(&luid, &ifindex) != NO_ERROR) {
+    logger.error() << "Failed to resolve GUID";
+    return false;
+  }
+
+  netsh.setProgram("netsh");
+  netsh.start();
+  if (!netsh.waitForStarted(WINDOWS_NETSH_TIMEOUT_MSEC)) {
+    logger.error() << "Failed to start netsh";
+    return false;
+  }
+
+  QTextStream cmdstream(&netsh);
+
+  // Flush DNS servers
+  QString v4flush = QString(netshFlushTemplate).arg("ipv4").arg(ifindex);
+  QString v6flush = QString(netshFlushTemplate).arg("ipv6").arg(ifindex);
+  logger.debug() << "netsh write:" << v4flush.trimmed();
+  cmdstream << v4flush;
+  logger.debug() << "netsh write:" << v6flush.trimmed();
+  cmdstream << v6flush;
+
+  // Add new DNS servers
+  for (const QHostAddress& addr : resolvers) {
+    const char* family = "ipv4";
+    if (addr.protocol() == QAbstractSocket::IPv6Protocol) {
+      family = "ipv6";
+    }
+    QString nsAddr = addr.toString();
+    QString nsCommand =
+        QString(netshAddTemplate).arg(family).arg(ifindex).arg(nsAddr);
+    logger.debug() << "netsh write:" << nsCommand.trimmed();
+    cmdstream << nsCommand;
+  }
+
+  // Exit and cleanup netsh
+  cmdstream << "exit\r\n";
+  cmdstream.flush();
+  if (!netsh.waitForFinished(WINDOWS_NETSH_TIMEOUT_MSEC)) {
+    logger.error() << "Failed to exit netsh";
+    return false;
+  }
+
+  return netsh.exitCode() == 0;
+}
+
+bool DnsUtilsWindows::restoreResolvers() {
+  if (m_luid == 0) {
+    return true;
+  }
+
+  QList<QHostAddress> empty;
+  if (m_setInterfaceDnsSettingsProcAddr != nullptr) {
```

Comment:
In the previous check, the order is the opposite. it was `if (a == nullptr)...` No needs to change.